Warning about ignored return value

We introduced a warning in https://github.com/tasts-robots/vulp/pull/30:

```
external/vulp/spine/AgentInterface.cpp:31:3: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
  ::ftruncate(file_descriptor, bytes);
  ^~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

@boragokbakan Can we check the return value, and throw a runtime error if this instruction failed?

--- a/spine/AgentInterface.cpp
+++ b/spine/AgentInterface.cpp
@@ -29,16 +29,16 @@ namespace vulp::spine {
 void allocate_file(int file_descriptor, int bytes) {
   struct ::stat file_stats;
 
-  if(::ftruncate(file_descriptor, bytes) < 0) {
+  if (::ftruncate(file_descriptor, bytes) < 0) {
     throw std::runtime_error("Error truncating file, errno is " +
                              std::to_string(errno));
   }
 
   ::fstat(file_descriptor, &file_stats);
   if (file_stats.st_size < bytes) {
-    throw std::runtime_error("Error allocating " + std::to_string(bytes) +
-                             " bytes in shared memory. Errno is : " +
-                              std::to_string(errno));
+    throw std::runtime_error(
+        "Error allocating " + std::to_string(bytes) +
+        " bytes in shared memory. Errno is : " + std::to_string(errno));
   }
 }
 
@@ -58,7 +58,8 @@ AgentInterface::AgentInterface(const std::string& name, size_t size)
       spdlog::error(
           "Cannot open shared memory \"{}\": file already exists. Is the spine "
           "already running? Did it not exit properly? If not, run "
-          "``./tools/bazelisk run " "@vulp//tools/shm:clean -- {}``",
+          "``./tools/bazelisk run "
+          "@vulp//tools/shm:clean -- {}``",
           name, name);
     } else {
       spdlog::error("Cannot open shared memory file: errno is {}", errno);

--- a/spine/AgentInterface.cpp
+++ b/spine/AgentInterface.cpp
@@ -28,11 +28,17 @@ namespace vulp::spine {
  */
 void allocate_file(int file_descriptor, int bytes) {
   struct ::stat file_stats;
-  ::ftruncate(file_descriptor, bytes);
+
+  if(::ftruncate(file_descriptor, bytes) < 0) {
+    throw std::runtime_error("Error truncating file, errno is " +
+                             std::to_string(errno));
+  }
+
   ::fstat(file_descriptor, &file_stats);
   if (file_stats.st_size < bytes) {
     throw std::runtime_error("Error allocating " + std::to_string(bytes) +
-                             " bytes in shared memory");
+                             " bytes in shared memory. Errno is : " +
+                              std::to_string(errno));
   }
 }
 

